src/main/scala/com/google/cloud/spark/bigquery/BigQueryInsertableRelation.scala

```diff
@@ -0,0 +1,61 @@
+package com.google.cloud.spark.bigquery
+
+import java.math.BigInteger
+
+import com.google.cloud.bigquery.{BigQuery, TableDefinition}
+import com.typesafe.scalalogging.StrictLogging
+import org.apache.spark.sql.sources.{BaseRelation, InsertableRelation}
+import org.apache.spark.sql.types.StructType
+import org.apache.spark.sql.{DataFrame, SQLContext, SaveMode}
+
+// Used only to insert data into BigQuery. Getting the table metadata (existence,
+// number of records, size, etc.) is done by API calls without reading existing
+// data
+case class BigQueryInsertableRelation(val bigQuery: BigQuery,
+                                      val sqlContext: SQLContext,
+                                      val options: SparkBigQueryOptions)
+  extends BaseRelation
```

Comment:
Why not check if the table exists in relationprovider and inherit from bigqueryrelation?

It seems like you are unnecessarily duplicating code

Reply:
BigQueryRelation holds a reference to a TableInfo member. When the table is inserted in many cases the table does not exist, so we cannot have an instance of this member (generated by the BQ client). Notice the different implementation and return value of getTable(), and the different implementation of schema()

Reply:
You could refactor that. Or have your own TableDefinition that is just a wrapper of TableId (and maybe schema):
TableInfo.newBuilder(TableId.of("foo","bar"), StandardTableDefinition.newBuilder().build).build

but I suppose the duplication is ok